airbyte-webapp/src/components/EntityTable/components/StreamStatusCell.tsx

```diff
@@ -0,0 +1,178 @@
+  isSyncing: boolean;
+}
+
+interface AirbyteStreamWithStatusAndConfiguration extends AirbyteStreamAndConfiguration {
+  config?: FakeStreamConfigWithStatus;
+}
+
+function filterStreamsWithTypecheck(
+  v: AirbyteStreamWithStatusAndConfiguration | null
+): v is AirbyteStreamWithStatusAndConfiguration {
+  return Boolean(v);
+}
+
+const generateFakeStreamsWithStatus = (
+  connection: WebBackendConnectionRead
+): AirbyteStreamWithStatusAndConfiguration[] => {
+  return connection.syncCatalog.streams
+    .map<AirbyteStreamWithStatusAndConfiguration | null>(({ stream, config }) => {
+      if (stream && config) {
+        return {
+          stream,
+          config: {
+            ...config,
+            status: connection.status,
+            latestSyncJobStatus: connection.latestSyncJobStatus,
+            scheduleType: connection.scheduleType,
+            latestSyncJobCreatedAt: connection.latestSyncJobCreatedAt,
+            scheduleData: connection.scheduleData,
+            isSyncing: connection.isSyncing || true,
+          },
+        };
+      }
+      return null;
+    })
+    .filter(filterStreamsWithTypecheck);
+};
+
+const getStatusForStream = (stream: AirbyteStreamWithStatusAndConfiguration): StatusType => {
+  if (stream.config && stream.config.selected) {
+    if (stream.config.status === "active" && stream.config.latestSyncJobStatus !== "failed") {
+      if (
+        // This can be undefined due to historical data, but should always be present
+        stream.config.scheduleType &&
+        !["cron", "manual"].includes(stream.config.scheduleType) &&
+        stream.config.latestSyncJobCreatedAt &&
+        stream.config.scheduleData?.basicSchedule?.units &&
+        stream.config.latestSyncJobCreatedAt * 1000 < // x1000 for a JS datetime
+          dayjs()
+            // Subtract 2x the scheduled interval and compare it to last sync time
+            .subtract(stream.config.scheduleData.basicSchedule.units, stream.config.scheduleData.basicSchedule.timeUnit)
+            .valueOf()
+      ) {
+        return "Behind";
+      }
+      return "On Track";
+    } else if (stream.config.latestSyncJobStatus === "failed") {
+      return "Error";
+    }
+  }
+  return "Disabled";
+};
+
+const sortStreams = (streams: AirbyteStreamWithStatusAndConfiguration[]): Record<StatusType, number> =>
+  streams.reduce(
+    (sortedStreams, stream) => {
+      sortedStreams[getStatusForStream(stream)]++;
+      return sortedStreams;
+    },
+    // This is the intended display order thanks to Javascript object insertion order!
+    { /* "Action Required": 0, */ Error: 0, Behind: 0, "On Track": 0, Disabled: 0 }
+  );
+
+const StreamsBar: React.FC<{ streams: AirbyteStreamWithStatusAndConfiguration[] }> = ({ streams }) => {
+  const sortedStreams = sortStreams(streams);
+  return (
+    <div className={styles.bar}>
+      {(Object.entries(sortedStreams) as Array<[StatusType, number]>)
+        .filter(([, count]) => !!count)
```

Comment:
Might be cleaner to extract this value into a var first

Reply:
I'm mixed on what I renamed this to.